### DISCUSSION TITLE
Automated Creation of Query Handlers for Domain Operations with Return Values

### DISCUSSION THREAD

**Discussion by stephanjohnson** [IntentArchitect/Support#46] [General]
I've come across a scenario that could potentially improve its functionality even further, particularly when dealing with complex aggregate root domain entities. Quite frequently, I find myself working with operations that require the return of data from aggregates nested within the aggregate root. Currently, it appears that Intent Architect doesn't automatically create query handlers for these specific types of domain operations.

As a potential enhancement, could the Project from Domain and the Create CQRS CRUD Operations context menu be extended to map these operations to query handlers? This feature would make handling complex domains a lot more manageable and expand the system's overall usefulness.

**Comment by joelsteventurner** [comment]
Hi @stephanjohnson 

I have added this to our backlog, another good suggestion.
